classes/Hook.php

```diff
@@ -582,6 +582,18 @@ public static function registerHook($module_instance, $hook_name, $shop_list = n
                 return false;
             }
 
+            // check that hook listener is implemented by the module
+            $expectedHookListenerMethod = sprintf('hook%s', ucfirst($hook_name));
```

Comment:
I think - I'm not sure - that hook callbacks are case-insensitive so we might need to check both `sprintf('hook%s', ucfirst($hook_name))` and `sprintf('hook%s', lcfirst($hook_name))` 🤔  ?

Reply:
They are ineed case-insensitive, you can have `hookdisplayprodctadditionalinfo` 👿 and it still does work

@matthieu-rolland maybe you could use this method:
https://github.com/PrestaShop/PrestaShop/blob/develop/classes/Hook.php#L386 ?

Reply:
Indeed, looks like the methods `getAllKnownNames` and `getHookAliasesFor` take care of the different possible hook names.

Fixed 👍